enzyme/Enzyme/Enzyme.cpp

```diff
@@ -23,6 +23,8 @@
 // the function passed as the first argument.
 //
 //===----------------------------------------------------------------------===//
+#include <optional>
```

Comment:
We should use the LLVM optional

Reply:
https://discourse.llvm.org/t/deprecating-llvm-optional-x-hasvalue-getvalue-getvalueor/63716/38